girder/web_client/src/views/widgets/ItemListWidget.js

```diff
@@ -140,6 +151,104 @@ var ItemListWidget = View.extend({
+     * the DOM reflows
+     */
+    scrollPositionObserver: function () {
+        // Set the default selected height for the selected item
+        const target = $('.g-item-list');
+        if (window.MutationObserver && target.length > 0) {
+            // Default items to monitor for the scroll position
+            const widgetcontainer = $('.g-hierarchy-widget-container');
+            // If the observer already exists disconnect it so it can be recreated
+            if (this.observer && this.observer.disconnect) {
+                this.observer.disconnect();
+            }
+
+            // Event handler for loading images declared once
+            const onLoadImage = (event) => {
+                if (this.observer) {
+                    this.centerSelected();
+                }
+            };
+
+            this.observer = new MutationObserver((mutations) => {
+                // for every mutation
+                _.each(mutations, (mutation) => {
+                    // If no nodes are added we can return
+                    if (!mutation.addedNodes) {
+                        return;
+                    }
+
+                    // for every added node
+                    _.each(mutation.addedNodes, (node) => {
+                        if (node.className && node.className.indexOf('g-item-list') !== -1) {
+                            // We want to do a onetime scroll to position if the screen is idle
+                            if (window.requestIdleCallback) {
+                                // Processing time is 2 seconds, but should finish much faster
+                                requestIdleCallback(() => {
+                                    this.centerSelected();
+                                }, { timeout: 2000 });
+                            } else {
+                                this.centerSelected();
+                            }
+                        }
+
+                        // For any images added we wait until loaded and rescroll to the selected
+                        if (_.isFunction(node.getElementsByTagName)) {
+                            $('img', node).on('load', onLoadImage);
+                        }
+                    });
+                });
+            });
+
+            // bind mutation observer to a specific element (probably a div somewhere)
+            this.observer.observe(target.parent()[0], { childList: true, subtree: true });
+
+            // Remove event listeners and disconnect observer
+            const unbindDisconnect = () => {
+                if (this.observer) {
+                    widgetcontainer.unbind('scroll.observerscroll');
+                    widgetcontainer.unbind('wheel.observerscroll');
+                    this.observer.disconnect();
+                    this.observer = null;
+                    this.tempScrollPos = undefined;
+                }
+            };
```

Comment:
Can we think of a way to include this in the tests?  It may be far too much work to be worth it.

Reply:
As an extra precaution I'm adding a click observer to unbindDisconnect as well for the case in which a user tries to select a different item while images are still loading.  The problem with the standard scrolling is creating that many items in Jasmine usually takes too long (or I would have to restructure how I create items) and I don't think I can trigger an actual scrollevent in a non-scrolling element.  I can easily trigger a click event like I am clicking or interacting and that can trigger the unbind.  That is something I have a test for.

Reply:
Understood.

You can create items from the rest endpoint directly (calling `girder.rest.restRequest` with a `POST item`, rather than mimicking user behavior), though that may also be slower than desired.